hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/sink/TestWriteMergeOnReadWithCompact.java

```diff
@@ -89,11 +95,86 @@ protected Map<String, String> getMiniBatchExpected() {
     return expected;
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void testCompactWithCommitTimeOrder(boolean useLegacyConfig) throws Exception {
```

Comment:
This is not where we should put the data set validation tests, maybe add tests in `TestInputFormat`.